Content.Server/GameObjects/EntitySystems/SuspicionRoleSystem.cs

```diff
@@ -0,0 +1,74 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using Content.Server.GameObjects.Components.Suspicion;
+using Content.Shared.GameObjects.Components.Suspicion;
+using Robust.Server.Interfaces.GameObjects;
+using Robust.Shared.GameObjects.Systems;
+using Robust.Shared.Interfaces.Network;
+
+namespace Content.Server.GameObjects.EntitySystems
+{
+    public class SuspicionRoleSystem : EntitySystem
+    {
+        private readonly HashSet<SuspicionRoleComponent> _antagonists = new HashSet<SuspicionRoleComponent>();
+
+        public void AddAntagonist(SuspicionRoleComponent role)
+        {
+            _antagonists.Add(role);
+        }
+
+        public void RemoveAntagonist(SuspicionRoleComponent role)
+        {
+            _antagonists.Remove(role);
+        }
+
+        public override void Shutdown()
+        {
+            base.Shutdown();
+            _antagonists.Clear();
+        }
+
+        public override void Update(float frameTime)
+        {
+            base.Update(frameTime);
+
+            if (_antagonists.Count == 0)
+            {
+                return;
+            }
+
+            var announceTo = new List<INetChannel>();
+
+            foreach (var role in ComponentManager.EntityQuery<SuspicionRoleComponent>())
+            {
+                if (!role.AnnounceTraitors)
+                {
+                    continue;
+                }
+
+                role.AnnounceTraitors = false;
+
+                if (!role.Owner.TryGetComponent(out IActorComponent actor))
+                {
+                    continue;
+                }
+
+                announceTo.Add(actor.playerSession.ConnectedClient);
+            }
+
+            if (announceTo.Count == 0)
+            {
+                return;
+            }
+
+            var antagonistIds = _antagonists.Select(t => t.Owner.Name).ToHashSet();
+            var message = new SuspicionAlliesMessage(antagonistIds);
+
+            foreach (var channel in announceTo)
+            foreach (var traitor in _antagonists)
```

Comment:
Disgusting.

Reply:
Oh god you can actually stack syntax like this infinitely...

Reply:
But of course:

![image](https://user-images.githubusercontent.com/10968691/91167179-c8d34b80-e6d3-11ea-94a6-a5684c9df0a2.png)

Reply:
is that from mafia smug...

Reply:
it is!!!!